tests/unit/backend/utils/field-validation/checkbox-validation.spec.ts

```diff
@@ -0,0 +1,294 @@
+import { ValidateFieldError } from 'src/app/modules/submission/submission.errors'
+import { ProcessedCheckboxResponse } from 'src/app/modules/submission/submission.types'
+import { validateField } from 'src/app/utils/field-validation'
+import {
+  BasicField,
+  CheckboxValidationOptions,
+  ICheckboxFieldSchema,
+} from 'src/types'
+
+import {
+  generateDefaultField,
+  generateNewCheckboxResponse,
+} from '../../helpers/generate-form-data'
+
+type MakeCheckboxFieldOptions = {
+  required?: boolean
+  othersRadioButton?: boolean
+  validateByValue?: boolean
+  ValidationOptions?: CheckboxValidationOptions
+}
+describe('Checkbox validation', () => {
+  const makeCheckboxField = (
+    fieldId: string,
+    fieldOptions: string[],
+    options?: MakeCheckboxFieldOptions,
+  ) => {
+    return {
+      ...generateDefaultField(BasicField.Checkbox, options),
```

Comment:
similar to comment in #1181, possible to extend the helper functions to accommodate your needs instead of defining new ones?